clients/unshaded/src/main/java/tachyon/client/BlockOutStream.java

```diff
@@ -58,12 +58,14 @@ public static BlockOutStream get(TachyonFile tachyonFile, WriteType opType, int
    */
   public static BlockOutStream get(TachyonFile tachyonFile, WriteType opType, int blockIndex,
       long initialBytes, TachyonConf tachyonConf) throws IOException {
-
-    if (tachyonFile.mTachyonFS.hasLocalWorker()
-        && tachyonConf.getBoolean(Constants.USER_ENABLE_LOCAL_WRITE,
-            Constants.DEFAULT_USER_ENABLE_LOCAL_WRITE)) {
+    if ((tachyonFile.mTachyonFS.hasLocalWorker()
+            && tachyonConf.getBoolean(Constants.USER_ENABLE_LOCAL_WRITE,
+            Constants.DEFAULT_USER_ENABLE_LOCAL_WRITE)
+            && tachyonConf.get(
```

Comment:
Below is what I get in the log if I remove this condition. We see that a remote node has been chosen by the MaxFree balance strategy. However we have a local worker, which has not been chosen because the free space is lower than the remote worker chosen. But because we don't have the condition we are trying to write with local stream on this remote worker... 

15/08/13 10:45:01 INFO : Connecting local worker @ remoteNode/xxx.xxx.xxx.xxx:29998
15/08/13 10:45:01 INFO : Writing with local stream. tachyonFile: /LC81850262014297LGN02.tar.gz, blockIndex: 0, opType: CACHE_THROUGH
15/08/13 10:45:01 INFO : Folder /mnt/ramdisk/tachyonworker/8 was created!
15/08/13 10:45:01 INFO : /mnt/ramdisk/tachyonworker/8/5368709120 was created! tachyonFile: /LC81850262014297LGN02.tar.gz, blockIndex: 0, blockId: 5368709120, blockCapacityByte: 536870912
15/08/13 10:45:01 WARN : Fail to cache for:
java.io.IOException: org.apache.thrift.TApplicationException: Internal error processing cacheBlock
        at tachyon.worker.WorkerClient.cacheBlock(WorkerClient.java:176)
        at tachyon.client.TachyonFS.cacheBlock(TachyonFS.java:248)
        at tachyon.client.LocalBlockOutStream.close(LocalBlockOutStream.java:164)
        at tachyon.client.FileOutStream.close(FileOutStream.java:118)
        at com.google.common.io.Closer.close(Closer.java:206)
        at tachyon.shell.TFsShell.copyPath(TFsShell.java:173)
        at tachyon.shell.TFsShell.copyFromLocal(TFsShell.java:142)
        at com.atos.client.tachyon.TachyonInit$class.copyFromLocal(TachyonInit.scala:81)
        at com.atos.client.api.FileTachyonApi.copyFromLocal(FileTachyonApi.scala:47)
        at com.atos.client.api.FileTachyonApi$$anonfun$6$$anonfun$apply$11$$anonfun$apply$17$$anonfun$apply$18.apply(FileTachyonApi.scala:216)
        at com.atos.client.api.FileTachyonApi$$anonfun$6$$anonfun$apply$11$$anonfun$apply$17$$anonfun$apply$18.apply(FileTachyonApi.scala:215)
        at scala.util.Try$.apply(Try.scala:191)
        at com.atos.client.api.FileTachyonApi$$anonfun$6$$anonfun$apply$11$$anonfun$apply$17.apply(FileTachyonApi.scala:215)
        at com.atos.client.api.FileTachyonApi$$anonfun$6$$anonfun$apply$11$$anonfun$apply$17.apply(FileTachyonApi.scala:204)
        at scala.Option.map(Option.scala:145)
        at com.atos.client.api.FileTachyonApi$$anonfun$6$$anonfun$apply$11.apply(FileTachyonApi.scala:203)
        at com.atos.client.api.FileTachyonApi$$anonfun$6$$anonfun$apply$11.apply(FileTachyonApi.scala:173)
        at scala.Option.map(Option.scala:145)
        at com.atos.client.api.FileTachyonApi$$anonfun$6.apply(FileTachyonApi.scala:172)
        at org.scalatra.ScalatraBase$class.org$scalatra$ScalatraBase$$liftAction(ScalatraBase.scala:270)
        at org.scalatra.ScalatraBase$$anonfun$invoke$1.apply(ScalatraBase.scala:265)
        at org.scalatra.ScalatraBase$$anonfun$invoke$1.apply(ScalatraBase.scala:265)
        at org.scalatra.ApiFormats$class.withRouteMultiParams(ApiFormats.scala:178)
        at com.atos.client.api.FileTachyonApi.withRouteMultiParams(FileTachyonApi.scala:47)
        at org.scalatra.ScalatraBase$class.invoke(ScalatraBase.scala:264)
        at com.atos.client.api.FileTachyonApi.org$scalatra$json$JsonSupport$$super$invoke(FileTachyonApi.scala:47)
        at org.scalatra.json.JsonSupport$$anonfun$invoke$1.apply(JsonSupport.scala:74)
        at org.scalatra.json.JsonSupport$$anonfun$invoke$1.apply(JsonSupport.scala:68)
        at org.scalatra.ApiFormats$class.withRouteMultiParams(ApiFormats.scala:178)
        at com.atos.client.api.FileTachyonApi.withRouteMultiParams(FileTachyonApi.scala:47)
        at org.scalatra.json.JsonSupport$class.invoke(JsonSupport.scala:68)
        at com.atos.client.api.FileTachyonApi.invoke(FileTachyonApi.scala:47)
        at org.scalatra.ScalatraBase$$anonfun$runRoutes$1$$anonfun$apply$8.apply(ScalatraBase.scala:240)
        at org.scalatra.ScalatraBase$$anonfun$runRoutes$1$$anonfun$apply$8.apply(ScalatraBase.scala:238)
        at scala.Option.flatMap(Option.scala:170)
        at org.scalatra.ScalatraBase$$anonfun$runRoutes$1.apply(ScalatraBase.scala:238)
        at org.scalatra.ScalatraBase$$anonfun$runRoutes$1.apply(ScalatraBase.scala:237)
        at scala.collection.immutable.Stream.flatMap(Stream.scala:458)
        at org.scalatra.ScalatraBase$class.runRoutes(ScalatraBase.scala:237)
        at org.scalatra.ScalatraServlet.runRoutes(ScalatraServlet.scala:49)
        at org.scalatra.ScalatraBase$class.runActions$1(ScalatraBase.scala:163)
        at org.scalatra.ScalatraBase$$anonfun$executeRoutes$1.apply$mcV$sp(ScalatraBase.scala:175)
        at org.scalatra.ScalatraBase$$anonfun$executeRoutes$1.apply(ScalatraBase.scala:175)
        at org.scalatra.ScalatraBase$$anonfun$executeRoutes$1.apply(ScalatraBase.scala:175)
        at org.scalatra.ScalatraBase$class.org$scalatra$ScalatraBase$$cradleHalt(ScalatraBase.scala:193)
        at org.scalatra.ScalatraBase$class.executeRoutes(ScalatraBase.scala:175)
        at org.scalatra.ScalatraServlet.executeRoutes(ScalatraServlet.scala:49)
        at org.scalatra.ScalatraBase$$anonfun$handle$1.apply$mcV$sp(ScalatraBase.scala:113)
        at org.scalatra.ScalatraBase$$anonfun$handle$1.apply(ScalatraBase.scala:113)
        at org.scalatra.ScalatraBase$$anonfun$handle$1.apply(ScalatraBase.scala:113)
        at scala.util.DynamicVariable.withValue(DynamicVariable.scala:58)
        at org.scalatra.DynamicScope$class.withResponse(DynamicScope.scala:80)
        at org.scalatra.ScalatraServlet.withResponse(ScalatraServlet.scala:49)
        at org.scalatra.DynamicScope$$anonfun$withRequestResponse$1.apply(DynamicScope.scala:60)
        at scala.util.DynamicVariable.withValue(DynamicVariable.scala:58)
        at org.scalatra.DynamicScope$class.withRequest(DynamicScope.scala:71)
        at org.scalatra.ScalatraServlet.withRequest(ScalatraServlet.scala:49)
        at org.scalatra.DynamicScope$class.withRequestResponse(DynamicScope.scala:59)
        at org.scalatra.ScalatraServlet.withRequestResponse(ScalatraServlet.scala:49)
        at org.scalatra.ScalatraBase$class.handle(ScalatraBase.scala:111)
        at org.scalatra.ScalatraServlet.org$scalatra$servlet$ServletBase$$super$handle(ScalatraServlet.scala:49)
        at org.scalatra.servlet.ServletBase$class.handle(ServletBase.scala:43)
        at com.atos.client.api.FileTachyonApi.org$scalatra$servlet$FileUploadSupport$$super$handle(FileTachyonApi.scala:47)
        at org.scalatra.servlet.FileUploadSupport$class.handle(FileUploadSupport.scala:93)
        at com.atos.client.api.FileTachyonApi.org$scalatra$CorsSupport$$super$handle(FileTachyonApi.scala:47)
        at org.scalatra.CorsSupport$$anonfun$handle$1.apply$mcV$sp(CORSSupport.scala:200)
        at org.scalatra.CorsSupport$$anonfun$handle$1.apply(CORSSupport.scala:190)
        at org.scalatra.CorsSupport$$anonfun$handle$1.apply(CORSSupport.scala:190)
        at scala.util.DynamicVariable.withValue(DynamicVariable.scala:58)
        at org.scalatra.DynamicScope$class.withResponse(DynamicScope.scala:80)
        at org.scalatra.ScalatraServlet.withResponse(ScalatraServlet.scala:49)
        at org.scalatra.DynamicScope$$anonfun$withRequestResponse$1.apply(DynamicScope.scala:60)
        at scala.util.DynamicVariable.withValue(DynamicVariable.scala:58)
        at org.scalatra.DynamicScope$class.withRequest(DynamicScope.scala:71)
        at org.scalatra.ScalatraServlet.withRequest(ScalatraServlet.scala:49)
        at org.scalatra.DynamicScope$class.withRequestResponse(DynamicScope.scala:59)
        at org.scalatra.ScalatraServlet.withRequestResponse(ScalatraServlet.scala:49)
        at org.scalatra.CorsSupport$class.handle(CORSSupport.scala:188)
        at com.atos.client.api.FileTachyonApi.handle(FileTachyonApi.scala:47)
        at org.scalatra.ScalatraServlet.service(ScalatraServlet.scala:54)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
        at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:769)
        at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:585)
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143)
        at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:577)
        at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:223)
        at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1125)
        at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:515)
        at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:185)
        at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1059)
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
        at org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:215)
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:97)
        at org.eclipse.jetty.server.Server.handle(Server.java:497)
        at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:311)
        at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:248)
        at org.eclipse.jetty.io.AbstractConnection$2.run(AbstractConnection.java:540)
        at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:610)
        at org.eclipse.jetty.util.thread.QueuedThreadPool$3.run(QueuedThreadPool.java:539)
        at java.lang.Thread.run(Thread.java:745)
Caused by: org.apache.thrift.TApplicationException: Internal error processing cacheBlock
        at org.apache.thrift.TApplicationException.read(TApplicationException.java:111)
        at org.apache.thrift.TServiceClient.receiveBase(TServiceClient.java:71)
        at tachyon.thrift.WorkerService$Client.recv_cacheBlock(WorkerService.java:280)
        at tachyon.thrift.WorkerService$Client.cacheBlock(WorkerService.java:266)
        at tachyon.worker.WorkerClient.cacheBlock(WorkerClient.java:169)
        ... 99 more